ARCHITECTURE.md

```
@@ -0,0 +1,47 @@
+# Architecture
+
+## Overview
+
+This repository follows a modular, maintainable architecture designed for clarity, scalability, and long-term sustainability.
+
+## Directory Structure
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)